### DISCUSSION TITLE
It's possible to integrate a debugger developed for VSCode in Atom?

### DISCUSSION THREAD

**Discussion by matteobertoldo** [matteobertoldo/atomforce#52] [Ideas]
In Salesforce [B2C Development](https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/ScriptDebuggerAPI/latest/usage/ScriptDebuggerAPI.html?cp=0_16) this feature is very important for the workflow.

Some [extensions](https://github.com/SqrTT/prophet/blob/master/src/Connection.ts) for VSCode have already integrated a [debugger](https://github.com/SqrTT/prophet/blob/master/src/mockDebug.ts) using the native API, it's possible with the Nuclide debugger or your Atom IDE community [debugger](https://github.com/atom-community/atom-ide-debugger) to be able to use the already written source code and integrate it into Atom?

If not, it's necessary need to develop an Atom-specific one instead using the debugger API? 
@aminya or @UziTech Can anyone help me? 🙏🏽 i don't know how to start 😅 I can sponsor you, no problem. 

<img width="1392" alt="Schermata 2021-01-22 alle 19 48 17" src="https://user-images.githubusercontent.com/15775323/105532409-15569d80-5ceb-11eb-8fee-7b8a71b67eee.png">

**Comment by UziTech** [comment]
There is a project to get VSCode extensions to work in Atom (https://github.com/atom-community/atom-vscode) but I don't think it is ready yet. I don't know a lot about the difference in the debugger APIs. @aminya should know more about that than me.

**Reply by matteobertoldo** [reply]
Thanks Tony! 🙏🏽

**Comment by aminya** [comment]
For debuggers, this is already possible, and we are doing it for JavaScript and Python. I have not found time to add other adapters, but other ones are the same.
https://github.com/atom-community/atom-ide-debugger/
https://github.com/atom-community/atom-ide-javascript/tree/master/src/debugger/node

What language do you want support for? 

Maybe, I should write a blog post on our website for adding adapters.

**Reply by matteobertoldo** [reply]
Thank you very much @aminya for the linked references . 🙏🏽

I think the debugger in question to be developed is executable via Node according to Salesforce documentation (see the links on the first post I wrote), in case here it is https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2FScriptDebuggerAPI%2Flatest%2Fusage%2FScriptDebuggerAPI.html&cp=0_16...
it should be only `https` requests and every request the server returns a trivial `JSON`...but I have no idea how to use the API, like creating a `launch`, an `attach`, `stepInto`, `stepOut` etc...

if you have time my friend, to write a post on how to create a custom debugger or in any case use the debuggers already written for VSCode...thank you very much.
If we manage to consolidate this ambitious goal, Atom becomes, undoubtedly, also for Salesforce B2C development an excellent alternative to other editors...because already with the UI that was created with this extension that I wrote years ago (to upload files and folders on the Salesforce WebDAV server) and an additional extension that highlights file syntax, remains the best editor of all....

**Reply by matteobertoldo** [reply]
Now I have created a new branch (`feature/matteobertoldo/debugger`) on which I can, at least try to do some small tests. 
For example, using the Nuclide super basic API I already see that the "Atomforce" debugger appeared in the debugger panel but from here, then I have no idea how to launch the debugger and just create scripts for which it can really be executed.

<img width="1392" alt="hero-debugger" src="https://user-images.githubusercontent.com/15775323/105584771-ce22e800-5d90-11eb-9c1c-a833a40be765.png">

**Reply by aminya** [reply]
Sorry for the late response. I lost the notification. I had to search in my emails to find the link for this repository. 

Is there a WIP branch of this code that you have prepared?

**Reply by matteobertoldo** [reply]
@aminya no prob! thanks for the response...the branch is `feature/matteobertoldo/debugger` (https://github.com/matteobertoldo/atomforce/tree/feature/matteobertoldo/debugger) i installed the `nuclide-debugger-common` pkg, and i added some super basic API (https://github.com/matteobertoldo/atomforce/blob/0e059e607df2773337ec447a9e64b456e3519c86/lib/atomforce.js#L64)
